components/hab/src/main.rs

```diff
@@ -706,6 +754,352 @@ fn sub_pkg_channels(ui: &mut UI, m: &ArgMatches) -> Result<()> {
+    let mut ui = ui();
+    let mut validate = protocol::ctl::SvcValidateCfg::new();
+    validate.set_service_group(service_group.clone().into());
+    let mut buf = Vec::with_capacity(protocol::butterfly::MAX_SVC_CFG_SIZE);
+    let cfg_len = match m.value_of("FILE") {
+        Some(f) => {
+            let mut file = File::open(f).unwrap();
+            file.read_to_end(&mut buf).unwrap()
+        }
+        None => io::stdin().read(&mut buf).unwrap(),
+    };
+    if cfg_len > protocol::butterfly::MAX_SVC_CFG_SIZE {
+        ui.fatal(format!(
+            "Configuration too large. Maximum size allowed is {} bytes.",
+            protocol::butterfly::MAX_SVC_CFG_SIZE
+        ))?;
+        process::exit(1);
+    }
+    validate.set_cfg(buf.clone());
+    let cache = default_cache_key_path(Some(&*FS_ROOT));
+    let mut set = protocol::ctl::SvcSetCfg::new();
+    match (service_group.org(), user_param_or_env(&m)) {
+        (Some(_org), Some(username)) => {
+            let user_pair = BoxKeyPair::get_latest_pair_for(username, &cache)?;
+            let service_pair = BoxKeyPair::get_latest_pair_for(&service_group, &cache)?;
+            ui.status(
+                Status::Encrypting,
+                format!(
+                    "TOML as {} for {}",
+                    user_pair.name_with_rev(),
+                    service_pair.name_with_rev()
+                ),
+            )?;
+            set.set_cfg(user_pair.encrypt(&buf, Some(&service_pair))?);
+            set.set_is_encrypted(true);
+        }
+        _ => set.set_cfg(buf.to_vec()),
+    }
+    set.set_service_group(service_group.into());
+    set.set_version(value_t!(m, "VERSION_NUMBER", u64).unwrap());
+    ui.begin(format!(
+        "Setting new configuration version {} for {}",
+        set.get_version(),
+        set.get_service_group(),
+    ))?;
+    ui.status(
+        Status::Creating,
+        format!("service configuration"),
+    )?;
+    SrvClient::connect(&sup_addr, &secret_key)
+        .and_then(|conn| {
+            conn.call(validate).for_each(
+                |reply| match reply.message_id() {
+                    "NetOk" => Ok(()),
+                    "NetErr" => {
+                        let m = reply.parse::<protocol::net::NetErr>().unwrap();
+                        match m.get_code() {
+                            ErrCode::InvalidPayload => {
+                                ui.warn(m)?;
+                                Ok(())
+                            }
+                            _ => Err(SrvClientError::from(m)),
+                        }
+                    }
+                    _ => {
+                        Err(SrvClientError::from(
+                            io::Error::from(io::ErrorKind::UnexpectedEof),
+                        ))
+                    }
+                },
+            )
+        })
+        .wait()?;
+    ui.status(
+        Status::Applying,
```

Comment:
This status might need to become `Setting`, depending on how we want to roll out the new emphasis on setting a config, rather than applying it.

Aaah... I see later on that this message is more about which supervisor you're connecting to; that was particularly confusing in the context of this particular command. This feels like a more appropriate use of the word "apply" (though we could perhaps use something like "Connecting")... we should probably accelerate the move to "set" when talking about configuration changes.

Incidentally, I notice that this `Status::Applying` isn't used in all commands... would it be worth including that feedback in all commands? If not, what's the criterion for inclusion/exclusion?

Reply:
We :100:% need to revisit the UX of the CLI. I didn't want to change the nomenclature in this PR